src/psd_tools/api/layers.py

```diff
@@ -812,7 +1013,79 @@ class PixelLayer(Layer):
         composed_image.save('composed-layer.png')
     """
 
-    pass
+    @classmethod
+    def frompil(cls, pil_im, psd_file, layer_name = "Layer", top = 0, left = 0, compression = Compression.RLE):
+        """
+        Method to create a layer from a PIL Image object, currently tuned for RGBA image.
+
+        :param pil_im: The :py:class:`~PIL.Image` object to convert to photoshop
+        :param layer_name: The name of the layer. Defaults to "Layer"
+        :param top: Pixelwise offset from the top of the canvas for the new layer.
+        :param left: Pixelwise offset from the left of the canvas for the new layer.
+        :param compression: Compression algorithm to use for the data.
+        :param psdfile: The psd file the image will be converted for.
+
+        :return: A :py:class:`~psd_tools.api.layers.PixelLayer` object
+     
+        """
+
+        assert pil_im
+
+        if pil_im.mode == "1":
```

Comment:
Yes, as long as the psd is provided it will convert the image to the proper mode. By the way, writing a psd with layers in other color modes than the PSD expects doesn't seems to hinder photoshop when opening the psd file. Photoshop itself will make the conversion (though i think its cleaner to force it on the images in the code)